src/Postal.AspNetCore/ServiceCollectionExtensions.cs

```diff
@@ -17,5 +19,60 @@ public static IServiceCollection AddPostal(this IServiceCollection services)
             services.TryAddScoped<IEmailViewRender, EmailViewRender>();
             return services;
         }
+
+        public static IServiceCollection ConfigurePostal(this IServiceCollection services, string smtpConnectionString)
+        {
+            var match = Regex.Match(smtpConnectionString, @"^smtp:\/\/((?<username>[^:@]+):(?<password>[^:@]+)@)?(?<host>[^:@,\/?#]+)(:(?<port>\d+))?(\?(?<option>[^&]+=[^&]+)(&(?<option>[^&]+=[^&]+))*)?$");
```

Comment:
This is a watered down version of the mongodb connection string parser

https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs#L764

Reply:
👍